src/prefect/serialization/flow.py

```diff
@@ -51,6 +51,7 @@ class Meta:
     )
     environment = fields.Nested(EnvironmentSchema, allow_none=True)
     storage = fields.Nested(StorageSchema, allow_none=True)
+    has_result_handler = fields.Boolean(allow_none=True)
```

Comment:
I think a better route might be to serialize the result handler directly, using the [`ResultHandlerSchema`](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/serialization/result_handlers.py#L91), and `allow_none=True`.  That way we can test for not-None in Cloud